src/executor.rs

```diff
@@ -25,161 +30,90 @@ pub struct Task {
     /// If `with_stdout` capture and send also its stdout.
     pub fn new(mission: &Mission) -> Result<Self> {
-        let mut command = mission.get_command();
+        let mut command = Command::from(mission.get_command());
         let with_stdout = mission.need_stdout();
-        let (task_sender, task_receiver) = bounded::<Task>(1);
-        let (stop_sender, stop_receiver) = bounded(0);
-        let (line_sender, line_receiver) = unbounded();
+        let (task_sender, mut task_receiver) = channel::<Task>(1);
+        let (stop_sender, mut stop_receiver) = oneshot::channel();
+        let (line_sender, line_receiver) = crossbeam::channel::unbounded();
         command
+            .stdin(Stdio::null())
             .stderr(Stdio::piped())
-            .stdout(if with_stdout { Stdio::piped() } else { Stdio::null() });
+            .stdout(if with_stdout {
+                Stdio::piped()
+            } else {
+                Stdio::null()
+            });
+
         let thread = thread::spawn(move || {
-            loop {
-                select! {
-                    recv(task_receiver) -> task => {
-                        let task = match task {
-                            Ok(task) => task,
-                            _ => { break; }
-                        };
-                        debug!("starting task {:?}", task);
-                        command.env(
-                            "RUST_BACKTRACE",
-                            if task.backtrace { "1" } else { "0" },
-                        );
-                        let child = command.spawn();
-                        let mut child = match child {
-                            Ok(child) => child,
-                            Err(e) => {
-                                if let Err(e) = line_sender.send(CommandExecInfo::Error(
-                                    format!("command launch failed: {}", e)
-                                )) {
-                                    debug!("error when sending launch error: {}", e);
-                                    break;
-                                }
-                                continue;
-                            }
-                        };
-                        let stderr = match child.stderr.take() {
-                            Some(stderr) => stderr,
-                            None => {
-                                if let Err(e) = line_sender.send(CommandExecInfo::Error(
-                                    "taking stderr failed".to_string()
-                                )) {
-                                    debug!("error when sending stderr error: {}", e);
-                                    break;
-                                }
-                                continue;
+            // start a runtime to manage the executor
+            let rt = tokio::runtime::Builder::new_current_thread()
+                .enable_io()
+                .build()
+                .unwrap();
```

Comment:
I believe it would be better if we use tokio in bacon to init it in main.

Reply:
Probably, but I didn't want to expand the scope of this PR more than necessary, so I kept it confined to this single file.